odl/discr/partition.py

```diff
@@ -0,0 +1,561 @@
+        ((True, False), (False, True))
+        """
+        bdry_list = []
+        for ax, (minpt, maxpt, bmin, bmax) in enumerate(zip(
+                self.grid.min_pt, self.grid.max_pt,
+                self.bbox.begin, self.bbox.end)):
+            bdry_list.append((minpt == bmin, maxpt == bmax))
+
+        return tuple(bdry_list)
+
+    def sampling_points(self):
+        """Return the sampling grid points."""
+        return self.grid.points()
+
+    def sampling_meshgrid(self):
+        """Return the sparse meshgrid of sampling points."""
+        return self.grid.meshgrid()
+
+    def cell_boundaries(self):
+        """Return the cell boundaries as coordinate vectors.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.cell_boundaries()
+        (array([-0.5,  0.5,  1.5]), array([-1.5, -0.5,  1. ,  3. ]))
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.cell_boundaries()
+        (array([ 0. ,  0.5,  1. ]), array([-1. , -0.5,  1. ,  2. ]))
+        """
+        bdry_vecs = []
+        for ax, vec in enumerate(self.grid.coord_vectors):
+            bdry = np.empty(len(vec) + 1)
+            bdry[1:-1] = (vec[1:] + vec[:-1]) / 2.0
+            bdry[0] = self.min()[ax]
+            bdry[-1] = self.max()[ax]
+            bdry_vecs.append(bdry)
+
+        return tuple(bdry_vecs)
+
+    def cell_sizes(self):
+        """Return the cell sizes as coordinate vectors.
+
+        Returns
+        -------
+        csizes : `tuple` of `numpy.ndarray`
+            The cell sizes per axis. The length of the vectors is the
+            same as the corresponding ``grid.coord_vectors``.
+            For axes with 1 grid point, cell size is set to 0.0.
+
+        Examples
+        --------
+        >>> grid = TensorGrid([0, 1], [-1, 0, 2])
+        >>> part = RectPartition(grid)  # Implicit begin and end
+        >>> part.cell_sizes()
+        (array([ 1.,  1.]), array([ 1. ,  1.5,  2. ]))
+
+        >>> part = RectPartition(grid, begin=[0, -1], end=[1, 2])
+        >>> part.cell_sizes()
+        (array([ 0.5,  0.5]), array([ 0.5,  1.5,  1. ]))
+        """
+        csizes = []
+        for ax, vec in enumerate(self.grid.coord_vectors):
+            if len(vec) == 1:
+                csizes.append(np.array([0.0]))
+            else:
+                csize = np.empty_like(vec)
+                csize[1:-1] = (vec[2:] - vec[:-2]) / 2.0
+                csize[0] = (vec[0] + vec[1]) / 2 - self.min()[ax]
+                csize[-1] = self.max()[ax] - (vec[-2] + vec[-1]) / 2
+                csizes.append(csize)
+
+        return tuple(csizes)
+
+    @property
+    def cell_size(self):
+        """Size of the 'inner' cells, regardless of begin and end.
+
+        Only defined if ``self.grid`` is a `RegularGrid`.
+
+        Examples
+        --------
+        >>> grid = RegularGrid([0, 0], [1, 1], (3, 5))
```

Comment:
When reading this, I feel most users would expect such a grid to have `as_midp=True` (in old terms), and thus that this guy has cell sizes (0.33, 0.2). Dont you agree?